git.sh

```diff
@@ -65,12 +65,21 @@ for commit in $commits; do
         echo Skipping $commit
     else
         echo "Running stats code for commit: $commit"
-        
+
         # Get the data to the specified commit
         cd data || exit $?
         git checkout $commit
         git clean -df
 
+        # skip this commit if there is no data since the later scripts fail if this is the case
+        directory_len=`ls -la | wc -l`
+        if [ $directory_len -lt 6 ]
```

Comment:
I might be missing something here, but how come this is less than `6`?

GIven this scenario:

```
$ mkdir testdir
$ cd testdir/
$ ls -la | wc -l
       3
$ touch iati_dataset.xml
$ ls -la | wc -l
       4
$ ls
iati_dataset.xml
```

Reply:
It's a git repository, so contains a `.git` folder. There's also another folder or file (can't recall without checking) that brings it up to 5... hence 6 being the minimum permitted.

Reply:
The conversation above demonstrates that 6 is [a magic number here](https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants). It shouldn’t be used without further explanation of its origin.